src/bezier/CubicBezierSolve.ts

```diff
@@ -0,0 +1,165 @@
+import Animated from "react-native-reanimated";
+
+import { find } from "../Array";
+import { acos } from "../Math";
+
+const {
+  Value,
+  and,
+  cond,
+  divide,
+  add,
+  multiply,
+  abs,
+  block,
+  greaterThan,
+  eq,
+  lessThan,
+  pow,
+  sqrt,
+  set,
+  sub,
+  cos,
+  not
+} = Animated;
+
+const isRootValidForCubicBezier = (root: Animated.Node<number>) =>
+  and(greaterThan(root, 0), lessThan(root, 1));
+
+// pomax.github.io/bezierinfo/#extremities
+const approximately = (
+  a: Animated.Adaptable<number>,
+  b: Animated.Adaptable<number>
+) => lessThan(abs(sub(a, b)), 0.001);
+
+const cuberoot = (v: Animated.Adaptable<number>) =>
+  cond(
+    lessThan(v, 0),
+    multiply(pow(multiply(v, -1), 1 / 3), -1),
+    pow(v, 1 / 3)
+  );
+
+const cubicBezierSolve = (
+  pa: Animated.Adaptable<number>,
+  pb: Animated.Adaptable<number>,
+  pc: Animated.Adaptable<number>,
+  pd: Animated.Adaptable<number>
+): Animated.Node<number> => {
+  const a: Animated.Value<number> = new Value();
+  const b: Animated.Value<number> = new Value();
+  const c: Animated.Value<number> = new Value();
+  const d: Animated.Value<number> = new Value();
+  const root1: Animated.Value<number> = new Value();
+  const root2: Animated.Value<number> = new Value();
+  const root3: Animated.Value<number> = new Value();
+
+  const q: Animated.Value<number> = new Value();
+  const q2: Animated.Value<number> = new Value();
+  const p: Animated.Value<number> = new Value();
+  const p3: Animated.Value<number> = new Value();
+  const discriminant: Animated.Value<number> = new Value();
+
+  const mp3: Animated.Value<number> = new Value();
+  const mp33: Animated.Value<number> = new Value();
+  const r: Animated.Value<number> = new Value();
+  const t: Animated.Value<number> = new Value();
+  const cosphi: Animated.Value<number> = new Value();
+  const phi: Animated.Value<number> = new Value();
+  const crtr: Animated.Value<number> = new Value();
+  const t1: Animated.Value<number> = new Value();
+
+  const u1: Animated.Value<number> = new Value();
+
+  const sd: Animated.Value<number> = new Value();
+  const v1: Animated.Value<number> = new Value();
+  const sq: Animated.Value<number> = new Value();
+
+  return block([
```

Comment:
👾 👾 👾

Reply:
I copied it line by line from https://pomax.github.io/bezierinfo/#quartic-curves-cardano-s-algorithm- works super well :) btw this page on bezier curves is amazing: https://pomax.github.io/bezierinfo. I learned so much from it, it's crazy

Reply:
I will dive into it through the nearest weekend, right after trying hard with `loop` :D

Reply:
haha `loop`, `toggle`,  and `delay`, are the real hard stuff :) I'm just distracting myself in between with some math stuff 😅🙈